build-args.cmake

```diff
@@ -4,6 +4,9 @@ else()
     set(LLAMA_METAL_DEFAULT OFF)
 endif()
 
+# general
+option(LLAMA_NATIVE "llama: enable -march=native flag" ON)
```

Comment:
I'm genuinely not sure why this caused issues.  If you have ideas or alternative approaches, do let me know.